### DISCUSSION TITLE
Maximum Number of Input Edges per Vertex

### DISCUSSION THREAD

**Discussion by andreasbinder** [pyg-team/pytorch_geometric#6145] [Q&A]
Hi all,
I was wondering what is the smoothest way to limit the number of input edges per vertex; I figured it might be integrated into the data loading part. Is  `NeighborLoader ` the correct way to do so?

```
loader = NeighborLoader(
    data,
    num_neighbors=[256], # lets say we want to restrict to maximum 256 edges per vertex
    ...
)
```

**Comment by wsad1** [comment]
Yes the approach you posted is a good way to limit the number of neighbors a nodes sees.

**Comment by LukeLIN-web** [comment]
When you set `num_neighbors=[256]`,  each vertex will sample 256 of its neighbor vertexes at its first hop.
